Runtime/AvatarCreator/UI/LoginElement.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using ReadyPlayerMe.Core;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.UI;
+
+namespace ReadyPlayerMe.AvatarCreator
+{
+    public class LoginElement : MonoBehaviour
+    {
+        private const string TAG = nameof(LoginElement);
+        [Header("Input Fields")]
+        [SerializeField, Tooltip("Input field for entering email address")] private InputField emailField;
+        [SerializeField, Tooltip("Input field for entering verification code")] private InputField codeField;
+
+        [Header("Events")]
+        [SerializeField] private UnityEvent OnLoginSuccess;
+        [SerializeField] private UnityEvent<string> OnLoginFail;
+
+        private void OnEnable()
+        {
+            AuthManager.OnSignInError += OnSignInError;
+        }
+
+        private void OnDisable()
+        {
+            AuthManager.OnSignInError -= OnSignInError;
+        }
+
+        public void SendVerificationCode()
+        {
+            AuthManager.SendEmailCode(emailField.text);
+        }
+
+        public async void LoginWithCode()
+        {
+            try
+            {
+                if (await AuthManager.LoginWithCode(codeField.text))
+                {
+                    OnLoginSuccess?.Invoke();
```

Comment:
Might it be a bit cleaner to keep consistency between the error/success approaches? Currently for 1 you have a separate method (error) and for the other you do it inline (success). I think either approach is fine since it's such a small thing, but I think consistency is the important thing here.

```suggestion
                    OnSignInSuccess();
                    
                    private void OnSignInSuccess() {
                       OnLoginSuccess?.Invoke();
                       SDKLogger.Log('Blah');
                    }
```

Reply:
updated and renamed some stuff